### PR TITLE
61 create indexjson and csv from file structure

### PR DESCRIPTION
**Pull request by NiklasTR**

### CODE REVIEW

**Comment by NiklasTR** [comment]
run the command from within the repo:

go run utils.go --application="diffdock" --input_directory="path/to/sdf-and-pdb"

**Review by thetechnocrat-dev — CHANGES_REQUESTED** [review]
It worked for me locally 💪 

There was a fair amount of scope creep in the PR around introducing a CLI and adding the concept of app config JSONL. Totally understand why that naturally emerged. Let's spend some engineering meeting time discussing a merge strategy.

Other smaller things
* remove dead code comments
* run `go fmt`
* break up large functions where possible
* less noise in code comments
* add unit tests (don't need 100% coverage but at least a few)
* I found the function names at the start of the utils file intuitive and then started to get confused by the function names towards the end of the file
* there are a lot of TODO comments

File: plexus/utils.go [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+// appStruct for the application file in the app.jsonl file
+//TODO #65 #64
+type appStruct struct {
+	App     string `json:"app"`
+	Inputs  [][2]string `json:"inputs"`
+	Outputs []string `json:"outputs"`
+}
+
+
+// validate that the directory path exists and is a directory
```
[the remark this answers is not in the document]
Comment (NiklasTR) [reply plexus/utils.go:28]: thank you - will make changes

File: plexus/utils.go [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+		new_files = append(new_files, prefix + "/" + filepath.Base(file))
+	}
+	print("Volume created: ", volume_path, "\n")
+	return id.String(), new_files, volume_path
+}
+
+// create a csv file that lists the indexed files in an application-specific format
+// the paths to the input files and the app config are given as input 
+// the path to the index.csv file is returned
+func indexCreateIndexJSONL(new_files []string, app_config *string, volume_path string) (string, []map[string]string) {
```
[the remark this answers is not in the document]
Comment (NiklasTR) [reply plexus/utils.go]: I broke it into three functions and renamed it

File: plexus/utils.go [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+		os.Exit(1)
+	}
+
+	for _, file := range files {
+		fmt.Println(file)
+	}
+	return files
+}
+
+func indexCreateInputsVolume(volume_directory *string, files []string, prefix string) (string, []string, string) {
```
[the remark this answers is not in the document]
Comment (NiklasTR) [reply plexus/utils.go]: resolved

**Comment by NiklasTR** [comment]
should be ready for sign-off generally. I will add tests tomorrow in the AM.

**Review by thetechnocrat-dev — APPROVED** [review]
🥇 LGTM

**Review by acashmoney — APPROVED** [review]
LGTM

**Pull request merged**